### PR TITLE
express-js -- store api

### PR DESCRIPTION
**Pull request by evilxgenius**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import usersRouter from './routes/users.routes.js';
+import productsRouter from "./routes/products.routes.js";
+import cartsRouter from "./routes/carts.routes.js";
+
+const app = express();
+const port = process.env.PORT || 3001;
+
+app.use(express.json());
+app.use(express.urlencoded({ extended: true }));
+app.use((req, _res, next) => (req.locals = {}) && next());
```
Comment (VitalyaIlin) [inline index.js:11]: why is this middleware needed?

File: index.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import express from 'express';
+import usersRouter from './routes/users.routes.js';
+import productsRouter from "./routes/products.routes.js";
+import cartsRouter from "./routes/carts.routes.js";
+
+const app = express();
+const port = process.env.PORT || 3001;
+
+app.use(express.json());
+app.use(express.urlencoded({ extended: true }));
```
Comment (VitalyaIlin) [inline index.js:10]: why is the `express.urlencoded` middleware needed?

File: models/user.model.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+    toJSON() {
+        return {
+            id: this.id,
+            email: this.email,
+            name: this.name
+        };
+    }
+
+    static async find(id) {
```
Comment (VitalyaIlin) [inline models/user.model.js:20]: why do you use `async` keyword here?

**Review by VitalyaIlin — COMMENTED** [review]
check comments please

File: index.js (hunk 1 above)
Comment (evilxgenius) [reply index.js:11]: The main idea is to exclude code duplication and write code more clearly. It initializes an additional field that I use for storing data that will added/used in the next middlewares or actions(get, post, etc.). For example, I get the cart object in .findCart middleware and it becomes available in the delete and put actions of cartRouter via req. locals.cart

File: index.js (hunk 2 above)
Comment (evilxgenius) [reply index.js:10]: This option (middleware) is for auto-parsing req.body. From express.js docs - A new body object containing the parsed data is populated on the request object after the middleware (i.e. req.body), or an empty object ({}). https://expressjs.com/en/5x/api.html#express.urlencoded:~:text=express.urlencoded(%5Boptions%5D)

**Pull request merged**